articles/142_idl.md

```diff
@@ -35,3 +35,45 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+IDL types mapped to different types in different languages.
+
+
+| IDL type | C type    | C++ type    | Python type          |
+| -------- | --------- | ----------- | -------------------- |
+| bool     | \_Bool    | bool        | bool                 |
+| octet    | uint8\_t  | uint8\_t    | bytes with length 1  |
+| char     | char      | char        | str with length 1    |
+| float    | float     | float       | float                |
+| double   | double    | double      | float                |
+| int8     | int8\_t   | int8\_t     | int                  |
+| uint8    | uint8\_t  | uint8\_t    | int                  |
+| int16    | int16     | int16       | int                  |
+| uint16   | uint16    | uint16      | int                  |
+| int32    | int32     | int32       | int                  |
+| uint32   | uint32    | uint32      | int                  |
+| int64    | int64     | int64       | int                  |
+| uint64   | uint64\_t | uint64\_t   | int                  |
+
+Unless are the default mappings for array, sequence, and string types unless otherwise specified.
+
+| IDL type                | C type                 | C++ type            | Python type |
+| ----------------------- | ---------------------- | ------------------- | ----------- |
+| array                   | T [FIXED\_SIZE]        | std::array<T, N>    | list        |
```

Comment:
IDL defines array as fixed size (7.4.1.4.4.5 `The array size (in each dimension) is fixed at compile time. The implementation of array indices is language mapping specific.`). `list` is a dynamic type in python. Maybe `tuple` makes more sense here? Only problem is when a user creates a message they can't append to the field, and instead must assign a new tuple too it. Passing a message with a tuple received on a subscription make sense though.